### PR TITLE
Implementation: star rating

### PR DESCRIPTION
**Pull request by kjniemela**
Ticket: https://trello.com/c/notjs3rd/80-shared-star-rating-component
added StarRating and Star components +tests

### CODE REVIEW

File: tests/shared.test.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  });
+});
+
+describe('Star component', () => {
+  test('a Star with with a rating fraction close to 1 should fill 100% of the star', () => {
+    const component = shallow( // mounts a `shallow` render of the component (does not render children)
+      <Star fraction={1} />
+    );
+    expect(component.containsMatchingElement(<stop offset='100%'></stop>)).toBe(true);
+    // expect(component.find('Star')).toHaveLength(5);
```
Comment (anerolabs) [inline tests/shared.test.js]: Tests look good-- curious though about the commented out assertion checking for number of stars in each test?

**Review by anerolabs — COMMENTED** [review]
Really like this implementation Johannes, clever use of building the fractions array and mapping over them!

File: tests/shared.test.js (hunk 1 above)
Comment (kjniemela) [reply tests/shared.test.js]: Ah! That's a mistake on my part, I had must have forgotten to remove it. It's not supposed to be there.

**Review by anerolabs — APPROVED** [review]
Looks good!

File: tests/shared.test.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  test('should have exactly 5 `Star` components as children', () => {
+    const component = shallow( // mounts a `shallow` render of the component (does not render children)
+      <StarRating rating={3} />
+    );
+    expect(component.find('Star')).toHaveLength(5);
+  });
+});
+
+describe('Star component', () => {
+  test('a Star with with a rating fraction close to 1 should fill 100% of the star', () => {
```
Comment (lbc1013) [inline tests/shared.test.js]: There's a duplicate word 'with'.

File: tests/shared.test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    const component = shallow( // mounts a `shallow` render of the component (does not render children)
+      <Star fraction={1} />
+    );
+    expect(component.containsMatchingElement(<stop offset='100%'></stop>)).toBe(true);
+  });
+
+  test('a Star with with a rating fraction close to 0.75 should fill 75% of the star', () => {
```
Comment (lbc1013) [inline tests/shared.test.js]: There's a duplicate word 'with'.

File: tests/shared.test.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    const component = shallow( // mounts a `shallow` render of the component (does not render children)
+      <Star fraction={0.75} />
+    );
+    expect(component.containsMatchingElement(<stop offset='75%'></stop>)).toBe(true);
+  });
+
+  test('a Star with with a rating fraction close to 0.5 should fill 50% of the star', () => {
```
Comment (lbc1013) [inline tests/shared.test.js]: There's a duplicate word 'with'.

File: tests/shared.test.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    const component = shallow( // mounts a `shallow` render of the component (does not render children)
+      <Star fraction={0.5} />
+    );
+    expect(component.containsMatchingElement(<stop offset='50%'></stop>)).toBe(true);
+  });
+
+  test('a Star with with a rating fraction close to 0.25 should fill 25% of the star', () => {
```
Comment (lbc1013) [inline tests/shared.test.js]: There's a duplicate word 'with'.

File: tests/shared.test.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    const component = shallow( // mounts a `shallow` render of the component (does not render children)
+      <Star fraction={0.25} />
+    );
+    expect(component.containsMatchingElement(<stop offset='25%'></stop>)).toBe(true);
+  });
+
+  test('a Star with with a rating fraction close to 0 should fill 0% of the star', () => {
```
Comment (lbc1013) [inline tests/shared.test.js]: There's a duplicate word 'with'.

**Review by lbc1013 — CHANGES_REQUESTED** [review]
It looks good except the comments that I wrote!

**Review by lbc1013 — APPROVED** [review]
👍

**Pull request merged**